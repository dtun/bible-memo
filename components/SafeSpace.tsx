import { View } from "react-native";
import {
  withSafeAreaInsets,
  WithSafeAreaInsetsProps,
} from "react-native-safe-area-context";

function SafeBottom(props: WithSafeAreaInsetsProps) {
  return <View style={{ height: props.insets.bottom }} />;
}

function SafeTop(props: WithSafeAreaInsetsProps) {
  return <View style={{ height: props.insets.top }} />;
}

export const SafeSpace = {
  Top: withSafeAreaInsets(SafeTop),
  Bottom: withSafeAreaInsets(SafeBottom),
};
