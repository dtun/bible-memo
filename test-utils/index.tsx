import { render } from "@testing-library/react-native";

const AllTheProviders = ({ children }: { children: React.ReactElement }) => {
  return <>{children}</>;
};

const customRender = (ui: React.ReactElement) =>
  render(ui, { wrapper: AllTheProviders });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
