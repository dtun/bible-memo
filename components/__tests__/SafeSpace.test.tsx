import * as React from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

import { render, screen } from "../../test-utils";
import { SafeSpace } from "../SafeSpace";

it(`renders <SafeSpace.Bottom />`, () => {
  render(
    <SafeAreaProvider>
      <SafeSpace.Bottom />
    </SafeAreaProvider>
  );

  expect(screen).toMatchSnapshot();
});

it(`renders <SafeSpace.Top />`, () => {
  render(
    <SafeAreaProvider>
      <SafeSpace.Top />
    </SafeAreaProvider>
  );

  expect(screen).toMatchSnapshot();
});
