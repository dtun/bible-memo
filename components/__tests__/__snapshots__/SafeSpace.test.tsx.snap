// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <SafeSpace.Bottom /> 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`renders <SafeSpace.Top /> 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
